docs: MVP plotly-express docs (#554)
deephaven/deephaven-plugins: plugins/plotly-express/docs/multiple-axes.md

# Multiple Axes

You can create multiple x or y axes in a single plot in a few different ways, from columns or partitions, or as layers from multiple plots. Multiple axis plots are useful for visualizing the relationship between variables that are expected to have some significant relationships, but have very different units or scales. In these cases, multiple axes can help display that relationship without forcing one variable to conform to the scale of the other.

In Deephaven Express, passing multiple columns to the `x` or `y` parameters along with setting a `yaxis_sequence` or `xaxis_sequence` creates multiple axes. Using the `by` parameter along with an axis sequence can create multiple axes, with one for each unique value in the column. The `layer` function can also be used to create multiple axes.

## Examples

### Multiple columns

> When two or more response variables appear in separate columns, passing multiple column names to `x` or `y` is the recommended way to create multiple axes.

Author: I don't understand, I get this:
<img width="867" alt="Screenshot 2024-07-26 at 11 48 41 AM" src="https://github.com/user-attachments/assets/abb90313-db62-4104-98fb-322aad18ff02">
which has the "pop" axis on the left and the "gdpPercap" axis on the right

> When two or more response variables appear in separate columns, passing multiple column names to `x` or `y` is the recommended way to create multiple axes.

Reviewer: Take this example:
```python
import deephaven.plot.express as dx
gapminder = dx.data.gapminder() # import a ticking version of the Gapminder dataset

# get a specific country
brazil = gapminder.where("country == `Brazil`")

# specify multiple y-axis columns and order axes left to right with yaxis_sequence
line_plot_multi = dx.line(brazil, x="year", y=["pop", "gdpPercap"], yaxis_sequence=[1, 2])
line_plot_multi_2 = dx.line(brazil, x="year", y=["pop", "gdpPercap"])
```

<img width="863" alt="image" src="https://github.com/user-attachments/assets/d2612b1e-4b37-4fc9-a24d-d223adddfd2f">
<img width="865" alt="image" src="https://github.com/user-attachments/assets/c078e033-3e2e-4026-9f98-2bc90bfe3c48">

Passing multiple value to y is **NOT** how the multiple axes are created.  That creates two lines.  To create separate axes for the lines, you need to specify `yaxis_sequence`.  This interaction is not clear in the prose or example.  The example would be better with two plots like I have.  Show that providing `y=[a,b]` gives two lines, and then adding `yaxis_sequence` puts them on different axes.  As it is, the prose is incorrect and doesn't show them how to do these two common cases.

```python order=line_plot_multi,brazil,gapminder
import deephaven.plot.express as dx
gapminder = dx.data.gapminder() # import a ticking version of the Gapminder dataset

# get a specific country
brazil = gapminder.where("country == `Brazil`")

# specify multiple y-axis columns and order axes left to right with yaxis_sequence
line_plot_multi = dx.line(brazil, x="year", y=["pop", "gdpPercap"], yaxis_sequence=[1, 2])
```

### The `by` parameter

When a single response variable has observations from several groups of data, use the `by` parameter to specify the grouping column.

```python order=line_plot_by,cat_dog,stocks
import deephaven.plot.express as dx
stocks = dx.data.stocks() # import the example stock market data set

# subset to get two symbols
cat_dog = stocks.where("sym in `CAT`, `DOG`")

# use `by` to specify the grouping column and order axes left to right with yaxis_sequence

> line_plot_by = dx.line(cat_dog, x="timestamp", y="price", by="sym", yaxis_sequence=[1, 2])

Author: In the particular example I wrote down, the stock data is always done deterministically, so the same data will always come first. In general you're right, may need a dict argument to be useful.

> line_plot_by = dx.line(cat_dog, x="timestamp", y="price", by="sym", yaxis_sequence=[1, 2])

Reviewer: This needs a ticket to assess the library.  It is not a problem with your example.

# Multiple Axes

Create plots with multiple axes by specifying `xaxis_sequence` or `yaxis_sequence`. Multiple axis plots are useful for visualizing the relationship between variables that have very different units or scales. In these cases, multiple axes can help display their relationship without forcing one variable to conform to the scale of the other.

## Examples

### Multiple columns

> When two or more response variables appear in separate columns, pass their column names to the `x` or `y` arguments and specify their order with `xaxis_sequence` or `yaxis_sequence.`
> 
> ```python order=line_plot_multi,brazil,gapminder
> import deephaven.plot.express as dx
> gapminder = dx.data.gapminder()
> 
> # get a specific country
> brazil = gapminder.where("Country == `Brazil`")
> 
> # specify multiple y-axis columns and split axes with yaxis_sequence
> line_plot_multi = dx.line(brazil, x="Year", y=["Pop", "GdpPerCap"], yaxis_sequence=[1, 2])
> ```
> 
> If `xaxis_sequence` or `yaxis_sequence` are not specified, the series will share an axis, which may or may not be useful depending on the units and scale of the data.
> 
> ```python order=line_plot_shared,brazil,gapminder
> import deephaven.plot.express as dx
> gapminder = dx.data.gapminder()
> 
> # get a specific country
> brazil = gapminder.where("Country == `Brazil`")
> 
> # population and per capita gdp have very different scales and units
> line_plot_shared = dx.line(brazil, x="Year", y=["Pop", "GdpPerCap"])
> ```

Reviewer: ```suggestion
When two or more response variables appear in separate columns, pass their column names to the `x` or `y` arguments.  The resulting chart will have shared axes.

```python order=line_plot_shared,brazil,gapminder
import deephaven.plot.express as dx
gapminder = dx.data.gapminder()

# get a specific country
brazil = gapminder.where("Country == `Brazil`")

# population and per capita gdp have very different scales and units
line_plot_shared = dx.line(brazil, x="Year", y=["Pop", "GdpPerCap"])
```

`xaxis_sequence` or `yaxis_sequence` can be used to create multiple axes.

```python order=line_plot_multi,brazil,gapminder
import deephaven.plot.express as dx
gapminder = dx.data.gapminder()

# get a specific country
brazil = gapminder.where("Country == `Brazil`")

# specify multiple y-axis columns and split axes with yaxis_sequence
line_plot_multi = dx.line(brazil, x="Year", y=["Pop", "GdpPerCap"], yaxis_sequence=[1, 2])
```
```

This is now in simple to complex order and doesn't try to imply usefulness.
